src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractHandlerProvider.cs

```diff
@@ -10,7 +10,11 @@ namespace Microsoft.CommonLanguageServerProtocol.Framework;
 /// <summary>
 /// Manages handler discovery and distribution.
 /// </summary>
+#if BINARY_COMPAT // TODO - Remove with https://github.com/dotnet/roslyn/issues/72251
 public abstract class AbstractHandlerProvider
+#else
+internal abstract class AbstractHandlerProvider
```

Comment:
The source package versions of these types need to be internal.  Otherwise someone referencing Clasp directly, and MS.CA.LanguageServer.Protocol (for example, Razor) will get conflicting type definitions if they're public.

However, for binary compatibility in VS on the old non-source package versions of clasp, these APIs need to be public.  So until we move all consumers to the source only version of the package, we have to ship the dll in VS with public accessibility.